### PR TITLE
SDS support for gloo

### PR DESCRIPTION
**Pull request by yuval-k**
part of #510

### CODE REVIEW

File: projects/gloo/api/v1/ssl.proto [hunk 1]
```diff
@@ -35,10 +37,41 @@ message SSLFiles {
...
     string sni = 3;
+
+    // Verify that the Subject Alternative Name in the peer certificate is one of the specified values.
+    // note that a root_ca must be provided if this option is used.
+    repeated string verify_subject_alt_name = 5;
+}
+
+message SDSConfig {
+    // Target uri for the sds channel. currently only a unix domain socket is supported.
+    string TargetUri = 1;
```
Comment (rickducott) [inline projects/gloo/api/v1/ssl.proto]: Should be snake case by convention

File: projects/gloo/api/v1/ssl.proto [hunk 2]
```diff
@@ -35,10 +37,41 @@ message SSLFiles {
...
+    // The name of the secret containing the certificate
+    string certificates_secret_name = 3;
+    // The name of secret containing the validation context (i.e. root ca)
+    string validation_context_name= 4;
 }
+
+message CallCredentials {
+    message FileCredentialSource {
+        // File containing auth token.
+        string TokenFileName = 1;
```
Comment (rickducott) [inline projects/gloo/api/v1/ssl.proto]: Should these be snake?

File: projects/gloo/pkg/utils/ssl.go [hunk 3]
```diff
@@ -69,6 +79,91 @@ func dataSourceGenerator(inlineDataSource bool) func(s string) *envoycore.DataSo
...
+					},
+				},
+			},
+		},
+	}
+}
+
+func (s *SslConfigTranslator) handleSds(sslSecrets *v1.SDSConfig, verifySan []string) (*envoyauth.CommonTlsContext, error) {
+	if sslSecrets.CertificatesSecretName == "" && sslSecrets.ValidationContextName == "" {
+		return nil, fmt.Errorf("at least one of certificates_secret_name or validation_context_name must be provided")
```
Comment (rickducott) [inline projects/gloo/pkg/utils/ssl.go]: Nit: is this or errors.errorf our convention?

File: projects/gloo/api/v1/ssl.proto (hunk 2 above)
Comment (yuval-k) [reply projects/gloo/api/v1/ssl.proto]: yes fixing snakes

File: projects/gloo/pkg/utils/ssl.go (hunk 3 above)
Comment (yuval-k) [reply projects/gloo/pkg/utils/ssl.go]: not sure i'll use errors.Errorf as ive seen it in the code

**Pull request merged**